hybrid/decomposers.py

```diff
@@ -599,3 +690,279 @@ def next(self, state, **runopts):
+                                                     target=target)
+    
+    #Remove minimal number of nodes such that edges yielded on subgraph:
+    G = nx.Graph()
+    G.add_edges_from(unyielded_edge_set)
+    for subgraph_nodes in nx.algorithms.components.connected_components(G):
+        cover = all_minimal_covers(G.subgraph(subgraph_nodes).edges)
+        for v in cover[0]:
+            del origin_embedding[v]
+    
+    #Restrict to giant component. 
+    proposed_source = proposed_source.subgraph(list(origin_embedding.keys()))
+    max_cc = max(nx.connected_components(proposed_source), key=len)
+    if len(max_cc) == 0:
+        raise ValueError('The proposed origin embedding contains no variables.' +
+                         + ' This is likely caused by a key error (not using'
+                         + ' geometrically appropriate variable keys.')
+    origin_embedding = {k:v for k,v in origin_embedding.items() if k in max_cc}
+    return origin_embedding
+
+def make_cubic_lattice(dimensions):
+    ''' Returns an open boundary cubic lattice graph as function of lattice
+    dimensions. Helper function for ``make_origin_embeddings`` '''
+    cubic_lattice_graph = nx.Graph()
+    cubic_lattice_graph.add_edges_from([((x, y, z),(x+1,y,z))
+                                    for x in range(dimensions[0]-1)
+                                    for y in range(dimensions[1])
+                                    for z in range(dimensions[2])])
+    cubic_lattice_graph.add_edges_from([((x, y, z),(x,y+1,z))
+                                    for x in range(dimensions[0])
+                                    for y in range(dimensions[1]-1)
+                                    for z in range(dimensions[2])])
+    cubic_lattice_graph.add_edges_from([((x, y, z),(x,y,z+1))
+                                    for x in range(dimensions[0])
+                                    for y in range(dimensions[1])
+                                    for z in range(dimensions[2]-1)])
+    return cubic_lattice_graph
+
+def make_origin_embeddings(qpu_sampler = None, lattice_type = None):
+    """Creates optimal embeddings for a lattice, 
+    compatible with the topology and shape of a ``qpu_sampler``. 
+
+    Args:
+        qpu_sampler (:class:`dimod.Sampler`, optional, default=\ :class:`~dwave.system.samplers.DWaveSampler`\ ``(client="qpu")``):
+            Quantum sampler such as a D-Wave system.
+
+        lattice_type (string, optional, default = qpu_sampler.properties['topology']['type']):
+            Options are:
+                * "cubic" 
+                    embeddings compatible with the schemes arXiv:2009.12479 and
+                    arXiv:2003.00133 are created for a ``qpu_sampler`` of 
+                    topology type either 'pegasus' or 'chimera'.
+
+                * "pegasus" 
+                    if ``qpu_sampler`` topology type is 'pegasus', maximum 
+                    scale nice subgraphs are embedded, with chain length one 
```

Comment:
"nice subgraphs"?

Reply:
pegasus_graph(nice_coordinates=True), particular subgraphs of pegasus. This an be conveyed more clearly.

Reply:
Embeddings are chain length one (minimal and native).
If ``qpu_sampler`` topology type is 'pegasus', maximum 
scale subgraphs are embedded using the ``nice_coordinates``
vector labeling scheme for variables.